esphome/components/mqtt/mqtt_climate.cpp

```diff
@@ -155,6 +159,12 @@ bool MQTTClimateComponent::publish_state_() {
     case CLIMATE_MODE_HEAT:
       mode_s = "heat";
       break;
+    case CLIMATE_MODE_FAN_ONLY:
+      mode_s = "fan";
```

Comment:
```suggestion
      mode_s = "fan_only";
```

The value is received here: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/mqtt/climate.py#L463 and the value should be `fan_only` as here https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/climate/const.py#L23

Also, MQTT doesn't appear to support fan mode property right now. It's fine for now, but it should be documented that it isn't supported right now (probably "climate core" page in docs is the best place).